app/src/main/java/org/jellyfin/mobile/utils/ApiExtensions.kt

```diff
@@ -0,0 +1,24 @@
+package org.jellyfin.mobile.utils
+
+import org.jellyfin.apiclient.interaction.ApiClient
+import org.jellyfin.apiclient.interaction.Response
+import org.jellyfin.apiclient.model.system.PublicSystemInfo
+import kotlin.coroutines.Continuation
+import kotlin.coroutines.resume
+import kotlin.coroutines.resumeWithException
+import kotlin.coroutines.suspendCoroutine
+
+// Can be removed/replaced once api client support coroutines natively
+suspend fun ApiClient.getPublicSystemInfo(): PublicSystemInfo? = suspendCoroutine { continuation ->
```

Comment:
I'm quite sure the return type won't be null since the onError will resume with an exception.

Reply:
Good catch, I actually wanted a `resume(null)` there.

Reply:
Can the exception even be nullable, btw?

Reply:
No but Kotlin makes all java types nullable unless they're annotated

Reply:
> No but Kotlin makes all java types nullable unless they're annotated

I know, that why I asked you :P